modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/episode/EpisodeFragmentViewModel.kt

```diff
@@ -120,7 +125,18 @@ class EpisodeFragmentViewModel @Inject constructor(
                     zipper,
                 )
             }
-            .doOnNext { if (it is EpisodeFragmentState.Loaded) { episode = it.episode } }
+            .doOnNext {
+                if (it is EpisodeFragmentState.Loaded) {
+                    episode = it.episode
+                    if (timestampInSecs != null &&
+                        it.episode.playedUpTo != timestampInSecs.toDouble() &&
```

Comment:
Floating point numbers shouldn't be compared directly due to numeric precision. It requires comparison with some epsilon.